### PR TITLE
Set Windows Terminal as pinned console

### PR DESCRIPTION
**Pull request by emtuls**
This sets Windows Terminal as the pinned console. It will also require merging the PR (https://github.com/mandiant/flare-vm/pull/580) in `flare-vm` that modifies the `CustomStartLayout.xml` to pin Windows Terminal instead of the command prompt.

I also had to modify `VM-Install-Shortcut` to allow for regular shortcuts to also include arguments with their execution, which was required for an admin Windows Terminal to be able to start in a directory that we want, since the "Working Directory" does not work for shortcuts that are set to run as admin.

### CODE REVIEW

File: packages/common.vm/tools/vm.common/vm.common.psm1 [hunk 1]
```diff
@@ -307,19 +307,22 @@ function VM-Install-Shortcut{
         if ($runAsAdmin) {
             $shortcutArgs.RunAsAdmin = $true
         }
-
         Install-ChocolateyShortcut @shortcutArgs
 
     } else {
+        if (-not $executableDir) {
```
Comment (Ana06) [inline packages/common.vm/tools/vm.common/vm.common.psm1:313]: what about moving this before the `if ($consoleApp -or $powershell)` instead of duplicating it?

File: packages/installer.vm/tools/chocolateyinstall.ps1 [hunk 2]
```diff
@@ -35,17 +35,10 @@ try {
...
-        $shortcut = Join-Path $shortcutDir "$toolName.lnk"
-        $workingDir  = Join-Path ${Env:UserProfile} "Desktop"
-        $arguments = "/k `"cd `"$workingDir`"`""
-
-        Install-ChocolateyShortcut -shortcutFilePath $shortcut -targetPath $executablePath -Arguments $arguments -RunAsAdmin
-        VM-Assert-Path $shortcut
+        # Create a Windows Terminal shortcut and it to pin to taskbar.
+        Write-Host "[+] Installing Windows Terminal..."
+        choco install windows-terminal.vm -y --force
+        refreshenv
```
Comment (Ana06) [inline packages/installer.vm/tools/chocolateyinstall.ps1:41]: why is this needed? I think we shouldn't ping it if the user didn't want to install it.

**Review by Ana06 — COMMENTED** [review]
I think we should remove the current `Cmd Here` and enable it in the Windows Terminal package in this PR. I think it should be run with admin rights, as we want admin rights to run malware.

File: packages/windows-terminal.vm/tools/chocolateyinstall.ps1 [hunk 3]
```diff
@@ -28,9 +28,48 @@ try {
...
+  # Add right click for Windows Terminal
+  $name = "Enable opening Windows Terminal from context menu - Default"
+  $path = "Registry::HKEY_CLASSES_ROOT\Directory\Background\shell\Windows Terminal"
+  $value = "(Default)"
+  $type = "String"
+  $data = "Open Windows Terminal here"  # This is how most programs word similar features
+  VM-Update-Registry-Value -name $name -path $path -value $value -type $type -data $data
+
+  $name = "Enable opening Windows Terminal from context menu - Default"
+  $path = "Registry::HKEY_CLASSES_ROOT\Directory\Background\shell\Windows Terminal"
```
[the remark this answers is not in the document]
Comment (emtuls) [reply packages/windows-terminal.vm/tools/chocolateyinstall.ps1:61]: You are correct! I was doing some copy and paste and it looked good visually, but I should have removed the redundancy 😂

File: packages/windows-terminal.vm/tools/chocolateyinstall.ps1 [hunk 4]
```diff
@@ -28,9 +28,48 @@ try {
...
   $dirList = Get-ChildItem $toolDir -Directory
   $toolDir = Join-Path $toolDir $dirList[0].Name -Resolve
-
+  $workingDir = Join-Path ${Env:UserProfile} "Desktop"
+  $arguments = "-p `"Command Prompt`" -d `"$workingDir`""   # Working directory doesn't work for admin shortcuts, so use -d flag for it.
   $executablePath = Join-Path $toolDir $executableName -Resolve
-  VM-Install-Shortcut -toolName $toolName -category $category -executablePath $executablePath -runAsAdmin
+  VM-Install-Shortcut -toolName $toolName -category $category -executablePath $executablePath -arguments $arguments -runAsAdmin
+
+  # Create a basic settings.json file so Windows Terminal always opens with elevated privileges
```
[the remark this answers is not in the document]
Comment (emtuls) [reply packages/windows-terminal.vm/tools/chocolateyinstall.ps1:36]: Hmmm. I'm not sure if I understand. The current method does make CMD the default. Does it not work that way for you?

[the remark this answers is not in the document]
Comment (Ana06) [reply packages/windows-terminal.vm/tools/chocolateyinstall.ps1:36]: I think I tested it wrong. I have tested again and it worked as I was proposing. When you open a new tab in the Terminal, it still defaults to PowerShell, but we can address that in a different PR. I have also created https://github.com/mandiant/VM-Packages/issues/971

**Review by Ana06 — APPROVED** [review]
Thanks for all the work @emtuls!
